### PR TITLE
Adds experiment tracker example to docs

### PR DESCRIPTION
**Pull request by RodolfoFerro**
### Details

I added a sample notebook (in `.md` format) to the docs, based on: https://ploomber.io/blog/experiment-tracking/

#### Solves: https://github.com/ploomber/ploomber-engine/issues/12

### CODE REVIEW

**Comment by RodolfoFerro** [comment]
**Update:** I tried to compile the docs directly from my dev branch and all changes seem to be ok, but for some reason an error is spotted (it already appeared in the last commit, prior to this one) -
https://readthedocs.org/projects/ploomber-engine/builds/18922498/

**Comment by edublancas** [comment]
by default readthedocs will fail if the docs fail with any warnings. the warning is this:

> Updating searchtools for Read the Docs search... [etoc] missing index.html written as redirect to 'intro.html'


I think if we rename intro.md to intro.html this will be fixed. please check

**Comment by RodolfoFerro** [comment]
Actually changing the file name caused the build to show this error:

```sh
Sphinx error:
root file /home/docs/checkouts/readthedocs.org/user_builds/ploomber-engine/checkouts/latest/doc/intro.rst not found
```

My second approach was trying this quickfix: https://github.com/executablebooks/jupyter-book/issues/1395#issuecomment-885937582

This last fix took me to find and solve some minor details to fix warnings and the compilation finished successfully: https://readthedocs.org/projects/ploomber-engine/builds/18933539/

No more problems were found, and the branch could now be merged without any problems. I already returned the rtd config for docs compilation directly from the main branch.

File: doc/environment.yml [hunk 1]
```diff
@@ -6,11 +6,15 @@ channels:
 
 dependencies:
   - python=3.10
+  - ipython
```
Comment (edublancas) [inline doc/environment.yml:9]: please comment that the newly added dependencies are required for the experiment tracker example. we might move this to a new package and when that happens we want to easily spot the dependencies that we don't need anymore, just add something like

```yaml
- scikit-learn # required for experiment tracker feature
```

**Review by edublancas — CHANGES_REQUESTED** [review]

File: doc/environment.yml (hunk 1 above)
Comment (RodolfoFerro) [reply doc/environment.yml]: Done!

**Review by edublancas — APPROVED** [review]

**Pull request merged**